stonesoup/types/orbitalstate.py

```diff
@@ -24,11 +24,15 @@ class CoordinateSystem(Enum):
     # To allow case insensitivity and the use of "TwoLineElement" as a string to mean TLE
     @classmethod
     def _missing_(cls, value):
-        for element in cls:
-            if element.value.lower() == value.lower():
-                return element
-            if element.value == "TLE" and value.lower() == "twolineelement":
-                return element
+        try:
+            for element in cls:
+                if element.value.lower() == value.lower():
+                    return element
+                if element.value == "TLE" and value.lower() == "twolineelement":
+                    return element
+
+        except ValueError:
+            raise ValueError("%r is not a valid %s" % (value, cls.__name__))
```

Comment:
No need for `try`/`except` here. Just have error raised at the end (as means other returns haven't been caught up).